Content.Client/Fluids/PuddleVisualizer.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Linq;
+using Content.Shared.Fluids;
+using JetBrains.Annotations;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Maths;
+using Robust.Shared.Random;
+using Robust.Shared.Serialization.Manager.Attributes;
+
+
+namespace Content.Client.Fluids
+{
+    [UsedImplicitly]
+    public class PuddleVisualizer : AppearanceVisualizer
+    {
+        [Dependency] private readonly IRobustRandom _random = default!;
+
+        // Whether the underlying solution color should be used
+        [DataField("recolor")] public bool Recolor;
+
+        public override void InitializeEntity(IEntity entity)
+        {
+            base.InitializeEntity(entity);
+
+            IoCManager.InjectDependencies(this);
+
+            var spriteComponent = entity.EnsureComponent<SpriteComponent>();
+            var maxStates = spriteComponent.BaseRSI?.ToArray() ?? Array.Empty<RSI.State>();
+
+            var variant = maxStates.Length switch
+            {
+                > 1 => _random.Next(0, maxStates.Length - 1),
+                1 => 0,
+                _ => 0,
+            };
```

Comment:
Can't you just do the random call directly? I doubt it's that much of a performance problem.